nilearn/glm/first_level/first_level.py

```diff
@@ -72,6 +73,26 @@ def _ar_model_fit(X, val, Y):
     return ARModel(X, val).fit(Y)
 
 
+def _yule_walker(X, order):
+    """Compute Yule-Walker (adapted from MNE and statsmodels).
+    Operates in-place.
+    """
+    from scipy import linalg
+    assert X.ndim == 2
+    assert order > 0, "AR order must be positive"
+    assert isinstance(order, int), "AR order must be an integer"
```

Comment:
We should test the function `_yule_walker` specifically. Although, I would replace the assertions with `ValueError` and `TypeError` exceptions and test that these are raised with bad arguments.

Reply:
Thanks for the suggestion. I add the errors and tests.

Reply:
Tested directly in 81b6d21b8d7e8f6d761a8dbecb15ab1e35a2fe62